src/components/ReservationModal.tsx

```diff
@@ -0,0 +1,93 @@
+import React, {useState} from 'react';
+import styled from 'styled-components';
+import { WHITE, BLACK } from '../colors';
+import { Modal } from 'antd';
+
+export enum ReservationModalType {
+  OPEN = 'OPEN',
+  TAKEN = 'TAKEN',
+  RESERVED = 'RESERVED'
+}
+
+interface ReservationModalProps {
+  readonly status: ReservationModalType;
+  readonly blockID: number;
+  readonly onOk: () => void;
+  readonly onCancel: () => void;
+  readonly isVisible: boolean;
+}
+
+const StyledModal = styled(Modal)`
+    height: 40px;
+    background: ${WHITE};
+    color: ${BLACK};
+    font-size: 5px;
+    font-weight: 400;
+`;
+
+const ReservationModal: React.FC<ReservationModalProps> = ({ 
+  status, blockID, onOk, onCancel, isVisible}) => {
+    const [visibility, setVisibility] = useState(isVisible);
+
+    const getOkText = (): string => {
+      switch (status) {
+        case ReservationModalType.OPEN : 
+          return 'Reserve'
+        case ReservationModalType.TAKEN : 
+          return 'Release'
+        case ReservationModalType.RESERVED : 
+          return 'Ok'
+      }
+    };
+
+    const handleCancel = (): void => {
+      onCancel();
+      setVisibility(false);
+    }
+
+    const handleOk = (): void => {
+      onOk();
+      setVisibility(false);
+    }
```

Comment:
These can be removed and we can have the modal below directly use onOk and onCancel after removing the local state of this component.